### PR TITLE
Refactor heartbeat logic

### PR DESCRIPTION
**Pull request by morningman**
Currently we only have Backend heartbeat. And without Frontend
or Broker heartbeat, we don't know the status of these nodes,
thus can't do failover logic in some cases.

1. Add Frontend and Broker heartbeat.
    Frontend heartbeat using BootstrapFinish http rest api
    Broker heartbeat using ping() rpc.
2. All heartbeats are managed in HeartbeatMgr.
3. Rename BrokerAddress to FsBroker.

ISSUE #402

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/catalog/FsBroker.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.system.BrokerHbResponse;
+import org.apache.doris.system.HeartbeatResponse.HbStatus;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+//Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/catalog/FsBroker.java]: license header should be before package and import

File: fe/src/main/java/org/apache/doris/system/Backend.java [hunk 2]
```diff
@@ -549,5 +526,48 @@ public DecommissionType getDecommissionType() {
         return DecommissionType.SystemDecommission;
     }
 
+    /*
+     * handle Backend's heartbeat response.
+     * return true if any port changed, or alive state is changed.
+     */
+    public boolean heartbeat(BackendHbResponse hbResponse) {
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/system/Backend.java]:
```suggestion
    public boolean handleHbResponse(BackendHbResponse hbResponse) {
```

File: fe/src/main/java/org/apache/doris/persist/HbPackage.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.system.HeartbeatResponse;
+
+import com.google.common.collect.Lists;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+//Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/persist/HbPackage.java]: license header position

**Review by imay — CHANGES_REQUESTED** [review]

File: fe/src/main/java/org/apache/doris/catalog/FsBroker.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/FsBroker.java]: Done

File: fe/src/main/java/org/apache/doris/system/Backend.java (hunk 2 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/system/Backend.java]: Done

File: fe/src/main/java/org/apache/doris/persist/HbPackage.java (hunk 3 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/persist/HbPackage.java]: Done

**Review by imay — APPROVED** [review]

**Pull request merged**